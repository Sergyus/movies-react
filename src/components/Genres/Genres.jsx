import React from 'react';
import PropTypes from 'prop-types';
import styles from './Genres.scss';

const Genres = (props) => {
  const { genres } = props;

  return (
    genres.map(genre => (
      <span key={genre.id} className={styles.genre}>
        {genre.name}
      </span>
    )));
};

Genres.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Genres;
