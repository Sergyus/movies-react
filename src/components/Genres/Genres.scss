.genre {
  margin-right: 3%;
}
