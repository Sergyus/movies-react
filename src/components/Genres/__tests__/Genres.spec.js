import React from 'react';
import { create } from 'react-test-renderer';
import Genres from '..';
import { genres } from '../../../assets/json/genres.json';

describe('Genres component renders correctly', () => {
  it('Genres renders correctly', () => {
    const tree = create(<Genres genres={genres} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
