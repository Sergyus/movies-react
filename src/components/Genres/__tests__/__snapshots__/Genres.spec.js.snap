// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres component renders correctly Genres renders correctly 1`] = `
Array [
  <span
    className="genre"
  >
    Action
  </span>,
  <span
    className="genre"
  >
    Adventure
  </span>,
  <span
    className="genre"
  >
    Animation
  </span>,
  <span
    className="genre"
  >
    Comedy
  </span>,
  <span
    className="genre"
  >
    Crime
  </span>,
  <span
    className="genre"
  >
    Documentary
  </span>,
  <span
    className="genre"
  >
    Drama
  </span>,
  <span
    className="genre"
  >
    Family
  </span>,
  <span
    className="genre"
  >
    Fantasy
  </span>,
  <span
    className="genre"
  >
    History
  </span>,
  <span
    className="genre"
  >
    Horror
  </span>,
  <span
    className="genre"
  >
    Music
  </span>,
  <span
    className="genre"
  >
    Mystery
  </span>,
  <span
    className="genre"
  >
    Romance
  </span>,
  <span
    className="genre"
  >
    Science Fiction
  </span>,
  <span
    className="genre"
  >
    TV Movie
  </span>,
  <span
    className="genre"
  >
    Thriller
  </span>,
  <span
    className="genre"
  >
    War
  </span>,
  <span
    className="genre"
  >
    Western
  </span>,
]
`;
